packages/gatsby-plugin-functions/src/gatsby-node.ts

```diff
@@ -67,51 +58,57 @@ export async function onPreBootstrap(
     }
   )
 
-  await Promise.all(
-    files.map(file => {
-      const config = {
-        entry: path.join(functionsDirectory, file),
-        output: {
-          path: path.join(process.cwd(), `.cache`, `functions`),
-          filename: file.replace(`.ts`, `.js`),
-          libraryTarget: `commonjs2`,
-        },
-        target: `node`,
-        // library: "yourLibName",
-
-        mode: `production`,
-        module: {
-          rules: [
-            {
-              test: [/.js$/, /.ts$/],
-              exclude: /node_modules/,
-              loader: `babel-loader`,
-            },
-          ],
-        },
-        plugins: [new webpack.DefinePlugin(varObject)],
-        // devtool: `source-map`,
-      }
-
-      return new Promise((resolve, reject) =>
-        // if (stage === `develop`) {
-        //   webpack(config).watch({}, () => {})
-
-        //   return resolve()
-        // }
-
-        webpack(config).run((err, stats) => {
-          console.log({
-            warnings: stats.compilation.warnings,
+  try {
+    await Promise.all(
+      files.map(file => {
+        const config = {
+          entry: path.join(functionsDirectory, file),
+          output: {
+            path: path.join(process.cwd(), `.cache`, `functions`),
+            filename: file.replace(`.ts`, `.js`),
+            libraryTarget: `commonjs2`,
+          },
+          target: `node`,
+          // library: "yourLibName",
+
+          mode: `production`,
+          module: {
+            rules: [
+              {
+                test: [/.js$/, /.ts$/],
+                exclude: /node_modules/,
+                loader: `babel-loader`,
+              },
+            ],
+          },
+          plugins: [new webpack.DefinePlugin(varObject)],
+          // devtool: `source-map`,
+        }
+
+        return new Promise((resolve, reject) =>
+          // if (stage === `develop`) {
+          //   webpack(config).watch({}, () => {})
+
+          //   return resolve()
+          // }
+
+          webpack(config).run((err, stats) => {
+            if (stats?.compilation?.warnings?.length > 0) {
+              reporter.warn(stats.compilation.warnings)
+            }
+
+            if (err) return reject(err)
+            const errors = stats.compilation.errors || []
+            if (errors.length > 0) return reject(stats.compilation.errors)
+            return resolve()
           })
-          if (err) return reject(err)
-          const errors = stats.compilation.errors || []
-          if (errors.length > 0) return reject(stats.compilation.errors)
-          return resolve()
-        })
-      )
-    })
-  )
+        )
+      })
+    )
+  } catch (e) {
+    reporter.error(`Failed to compile Gatsby Functions.`)
+    reporter.panicOnBuild(e)
```

Comment:
Any reason why you only panic on build and not in develop?
```suggestion
    activity.panicOnBuild(`Failed to compile Gatsby Functions.`, e)
```

Reply:
What should it be instead if you want to panic in both?

Reply:
just panic :)